### ISSUE TITLE
Rule proposal: no-unnecessary-promise-all for single promises

### ISSUE THREAD

**Issue by kkmuffme** [sindresorhus/eslint-plugin-unicorn#1938]
### Description

Report unnecessary calls to Promise.all(Settled) when there's just a single promise

### Fail

```js
Promise.all([promise1]).then((values) => {
  console.log(values);
});
```

```js
Promise.allSettled([promise1]).then((values) => {
  console.log(values);
});
```

### Pass

```js
Promise.all([promise1, promise2, promise3]).then((values) => {
  console.log(values);
});
```

```js
Promise.allSettled(promises).then((values) => {
  console.log(values);
});
```

**Comment by fisker** [comment]
Should we forbid `Promise.any()` and `Promise.race()` on single promise too?

**Comment by fisker** [comment]
Thanks, didn't notice #2256 is a duplicate of this.
